File: search-insert-position.java
import java.util.*;

// https://leetcode.com/problems/search-insert-position

class Solution {
    public int searchInsert(int[] nums, int target) {
        int index = 0;
        int length = nums.length;

        for (int i = 0; i < length; i++) {
            if (nums[i] >= target) return index;
            index++;
        }
        return index;
    }
}

File: search-insert-position.kt
class Solution {
    fun searchInsert(nums: IntArray, target: Int): Int {
        return binarySearch(nums, target)
    }

    fun binarySearch(nums: IntArray, target: Int): Int {
        var start = 0
        var end = nums.size - 1
        var mid: Int
        var answer = nums.size
        while (start <= end) {
            mid = (start + end) / 2
            if (nums[mid] >= target) {
                answer = mid
                end = mid - 1
            } else {
                start = mid + 1
            }
        }
        return answer
    }
}
